Islandora-CLAW/islandora_demo
# Issue 1065

**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/1065

* Other Relevant Links: Goes with https://github.com/Islandora-CLAW/islandora/pull/123

# What does this Pull Request do?

Consolidates repeated config in multiple contexts into a single context.  Also adds the AlterJSONLDType reaction to all media and content so that field_model and field_media_use have their external uris appended to rdf:type.

# How should this be tested?

- View a media and/or node's ?_format=jsonld.  The rdf:type will be either pcdm:File or pcdm:Object for media and nodes respectively.
- Pull in the associated pull for the islandora module: https://github.com/Islandora-CLAW/islandora/pull/123.
- `$ drush cr`
- Pull in this PR
- `drush fim -y islandora_demo`
- Now go view a media or node.  It should have the external URI for its model or use appended to rdf:type.  For example for a Media that's a Service File
```
{
   "@graph":[
      {
         "@id":"http:\/\/localhost:8000\/media\/2?_format=jsonld",
         "@type":[
            "http:\/\/pcdm.org\/models#File",
            "http:\/\/pcdm.org\/use#ServiceFile"
         ],
        ...
``` 
And for a node that's an Image
```
{
  "@graph": [
    {
      "@id": "http://localhost:8000/node/1?_format=jsonld",
      "@type": [
        "http://pcdm.org/models#Object",
        "http://purl.org/coar/resource_type/c_c513"
      ],
     ...
```

# Additional Notes:
There's some cleanup in there, too.  I'm honestly not sure if we were even indexing all Media in the triplestore to begin with.  Hopefully this consolidated config will help keep things from drifting.

# Interested parties
@ajstanley @Islandora-CLAW/committers
